precommits/pre-commit

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
```

Comment:
### Statement
I think that a pre-commit is a bit too strict as a constraints, it's fine to commit non linted code IMO. 

### Example
I can imagine the use case of a partial commit that you will make on your staging area in order to fix a stable version of your code while developing and later on rebase and eventually `amend` that commit to give a better explaination to the reviewer, i don't see the needs in that case to have a that level of strictness.

### Proposed alternative solution

- `pre-push`

Reply:
+1! It happened to me in other projects. it should be at a pre-push level so you can work regularly in intermediate commits avoiding the `--no-verify`.